include/dslash_quda.h

```diff
@@ -522,6 +523,81 @@ namespace quda
       const CloverField &C, double a, double b, bool xpay, const ColorSpinorField &x, int parity, bool dagger,
       const int *comm_override, TimeProfile &profile);
 
+  /**
+     @brief Driver for applying the non-degenerate twisted-clover
+     stencil
+
+     out = a * D * in + (C + i*b*gamma_5*tau_3 + c*tau_1) * x
+
+     where D is the gauged Wilson linear operator. The quark fields
+     out, in and x are five dimensional, with the fifth dimension
+     corresponding to the flavor dimension.  The convention is that
+     the first 4-d slice (s=0) corresponds to the positive twist and
+     the second slice (s=1) corresponds to the negative twist.
+
+     This operator can be applied to both single parity
+     (4d checker-boarded) fields, or to full fields.
+
+     @param[out] out The output result field
+     @param[in] in The input field
+     @param[in] U The gauge field used for the operator
+     @param[in] C The clover field used for the operator
+     @param[in] a Scale factor applied to Wilson term (typically -kappa)
+     @param[in] b Chiral twist factor applied (typically 2*mu*kappa)
+     @param[in] c Flavor twist factor applied (typically -2*epsilon*kappa)
+     @param[in] x Vector field we accumulate onto to
+     @param[in] parity Destination parity
+     @param[in] dagger Whether this is for the dagger operator
+     @param[in] comm_override Override for which dimensions are partitioned
+     @param[in] profile The TimeProfile used for profiling the dslash
+  */
+  void ApplyNdegTwistedClover(ColorSpinorField &out, const ColorSpinorField &in, const GaugeField &U,
+                              const CloverField &C, double a, double b, double c, const ColorSpinorField &x, int parity,
+                              bool dagger, const int *comm_override, TimeProfile &profile);
+
+  /**
+     @brief Driver for applying the non-degenerate preconditioned twisted-clover stencil
+
+     out = a * (C + i*b*gamma_5*tau_3 - c*tau_1)^{-1} * D * in + x
+         = a * (C - i*b*gamma_5*tau_3 + c*tau_1)/(C^2 + b^2 - c^2) * D * in + x
+         = A^{-1} * D * in + x
+
+     where D is the gauged Wilson linear operator and C is the clover
+     field.  This operator can (at present) be applied to only single
+     parity (checker-boarded) fields.  When the dagger operator is
```

Comment:
While I certainly appreciate the logic noted below (kernel fusion), `dagger` not meaning `dagger` does bother me---in particular I think it implies `DiracTwistedClover::MdagM` isn't a Hermitian positive-definite operator in the non-degenerate case?

To be clear, I'm not saying "throw out kernel fusion" because obviously this workflow is working, but I'm wondering if there's a way we can change the plumbing so `dagger` means `dagger` and the fused "dagger without transpose" operator is exposed another way.

Reply:
`MdagM` is a Hermitian operator for ndeg TMC.  It's just not composed using `M`, followed by `Mdag`.  This is not unique to nondeg TMC, and we've got this style for other operators.

Reply:
That's not what it looks like in the body of `DiracTwistedClover::MdagM`---it looks like it's indeed the sequence of `M` then `Mdag`, and I don't see a custom implementation of `Mdag`.

Reply:
Oops, my brain cross wires there.  Agreed, it is written as `M` and `Mdag` product.

Reply:
Correct what I have from `DiracTwistedClover` to `DiracTwistedCloverPC` --- I think `DiracTwistedClover` itself is fine.

Reply:
Let me try again: the operator `MdagM` is Hermitian, and the components `M` and `Mdag` are what they claim to be.  The issue is just that the components are put together to enable kernel fusion.

Reply:
And this isn't an issue, it's just that for kernel fusion, all operators must appear to the left of D_4, so when we call something daggered, we only mean we are daggering the operator components, and not the entire operator (e.g., reversing its order).  That's the convention that we've had for a long time.

Reply:
Where my brain crossed wires is that we could fuse across `M` / `Mdag` boundaries in order to give more latency hiding, in the same way that's already been done for DWF-4d / Möbius fermions.  But that optimization lives in my head only (a subtle, but important distinction 😉 ).

Reply:
Okay, I see where I went astray. I was comparing against `DiracCloverPC` where the mathematical dagger vs non-dagger is in `DiracCloverPC::M`, so when I didn't see that pattern in `DiracTwistedCloverPC` I got confused. What I missed is that pattern is in `DiracTwistedCloverPC::Dslash` instead.

tl;dr I was thrown off by an inconsistency of convention between regular clover and twisted clover, and now everything's clear.

Reply:
All is good.  This discussion reminded me of wanting to fuse across the Mdag / M boundaries for symmetric clover preconditioned kernels, so it was worthwhile for me 😄